src/components/gltf-model-plus.js

```diff
@@ -449,6 +468,38 @@ class GLTFHubsPlugin {
         }
 
         node.extras.gltfIndex = i;
+
+        if (qsTruthy("newLoader")) {
+          /**
+           * This guarantees that components that add Object3Ds (ie. through addObject3DComponent) are not attached to non-Object3D
+           * entities as it's not supported in the BitECS loader.
+           * This was supported by the AFrame loader so this extension ensures backwards compatibility with all the existing scenes.
+           * For more context about this see: https://github.com/mozilla/hubs/pull/6121
+           */
+          if (node.mesh !== undefined || node.camera !== undefined) {
+            const exts = node.extensions?.MOZ_hubs_components;
+            if (exts) {
+              const children = [];
+              for (const [key, value] of Object.entries(exts)) {
+                if (OBJECT3D_EXT.includes(key)) {
```

Comment:
nit: If `OBJECT3D_EXT` is used only for checking whether `key` is in a set, wouldn't using `Set` be more efficient than using array? The number of `OBJECT3D_EXT` entries is just 16, so probably not a big deal tho. Just a comment for minor optimization.